packages/babel-generator/src/generators/statements.js

```diff
@@ -123,10 +123,14 @@ function buildLabelStatement(prefix, key = "label") {
     const label = node[key];
     if (label) {
       this.space();
-
-      const terminatorState = this.startTerminatorless();
-      this.print(label, node);
-      this.endTerminatorless(terminatorState);
+      if (prefix == "break" || prefix == "continue") {
+        // see https://git.io/vQ7uG
+        this.print(label, node);
```

Comment:
> what do you think about converting single line comments

If it's something we can do, go for it. It might still not be easy. There's still the issue of multiline comments, e.g.

```
continue /* foo
bar */ label;
```

but as with my previous comment, either we throw a clear error, or maybe extending your approach, we also just replace newlines in the comments with spaces?